pkg/template/template.go

```diff
@@ -72,7 +73,9 @@ func (p *Processor) Process(template *api.Template) field.ErrorList {
 func stripNamespace(obj runtime.Object) {
 	// Remove namespace from the item
 	if itemMeta, err := meta.Accessor(obj); err == nil {
```

Comment:
if you do itemMeta.SetNamespace("") it will set the namespace to empty string and it will serialize it that way... we don't want to have empty namespace field in output, so we check if the namespace is set before resetting it to empty string.